### ISSUE TITLE
Decide on a concept behind a tech tree

### ISSUE THREAD

**Issue by cit1zen** [cit1zen/workers-and-resources-community-tech-tree#5]
Decide on a document and create a design document that would guide us when deciding which and where the technologies should be in the tech tree.

**Comment by argentlike** [comment]
[Techtree 0.1.xlsx](https://github.com/cit1zen/workers-and-resources-community-tech-tree/files/4361694/Techtree.0.1.xlsx)

**Comment by argentlike** [comment]
oops didnt mean to close it

**Comment by argentlike** [comment]
On discord WRSR you asked me to consider structure/flow and balance. I guess my first thoughts (and putting myself in our mod members shoes) was 'why'? Why should these things be connected. So I started checking on tech research streams and thought that we may want to consider a rationale, the linking story behind our tech tree. Then when modders want to attach prequisites and unlock they can do it hopefully a more informed way that will provide a natural balancing force. The tech tree needs to sit within WRSR dev vanilla, and within the techtree you have made a great start with - but we need a 'backstory'
I have done a first draft at that. And would share it. It is a placeholder approach, so dont get hung up on tech logic and what should go where. The base concept is we have some tech families and then some tech streams. It gives us a feel for where/when to attach our unlock buildings etc. I have suggested buildings, transport, energy etc as you will see. If we can agree on this, then writing the readme has enormous depth (apart from being very easy). Obviously we make the structure open - and we are not tied to what Peter decides - we place vanilla into our story. Yes, we will get forced along certain things, but it is a better method to justify ourselves.

It is designed on what you commented on, and I also searched and found similar ideas on our discord. For example some good thinking around 'blocks' which I have renamed streams.
I can just see the community come alive 'discussing' it and where their building should go.
Here is my first draft attempt we can butcher and rebuild.
I think we should take it to a first conceptual phase and then go for comment. much like a project approach, otherwise to many cooks and we will get nowhere. i think this will get everyone juices fired up - which is good, and bad if we want efficiency

**Comment by cit1zen** [comment]
My concept behind the tech tree:

## Core ideas and principles

The tech tree is based around several core ideas and principles that should guide its design and balance.

The first idea is that we do not research new technologies. It does not make sense that a small republic with 500 people would be able to research computers on its own. Instead, by "researching," we mean that our scientists travel to foreign lands and try to get experience, copy blueprints, acquire technologies, and learn new processes so that they can introduce them in our republic. That should explain why we are able to buy goods but are not able to build a factory that produces them.

The second core idea is about where our republics are when starting the game. Let's assume that there was no advanced industry before the revolution or colonization. That means that our people are able to build clothes or food factory, but should not be able to construct and manage a chemical plant.

The design principle is that the research tree should not be flat and that there should be some logical and realistic progression. That means that the republic should not be able to enrich uranium if it is unable to produce chemicals.

**Comment by argentlike** [comment]
_The second core idea is about where our republics are when starting the game._ 
Hmm, might need a full mod approach later, but the ability to choose in settings (like in 1960 or 1970 start)and gives you a diff starting tech position?

**Comment by cit1zen** [comment]
I created a second point so we have a better idea of a baseline, i.e., what should be available without research.

I think we can reconsider that point when better modding API is available (the mod approach you mentioned). Should we put it somewhere? The task to reconsider that point?

**Comment by argentlike** [comment]
Agreed, we should put it in a task so we dont lose it.  Not sure how this github structure works, you do it and I will pick things up as we go...
Your other comment about vanilla start seems to be what you have kicked off on a new discord channel.  btw, is that open or by invite?

**Comment by cit1zen** [comment]
It should be open. If not I will invite you.

**Comment by argentlike** [comment]
So, I started our prototype design doc.  We are agreed on the family & stream approach I think.  Suggestion:  We should develop one family & its streams as our first design doc task.  With that we can attach diff assets and see what issues we find in the proposed design structure.

**Comment by cit1zen** [comment]
I think we can and should have conversation about the families themselves.
I like the idea, but let's decide what those families are.

**Comment by argentlike** [comment]
Cool, so using the initial version design doc as a starting point, how do we need to alter the family choice?

**Comment by cit1zen** [comment]
I would like to see these families (most of them are in some way in your doc):
* Chemistry (chemicals, plastics)
* Mechanical engineering (mech components, cars, ships, trains)
* Electronics
* Extraction (mines, oil wells)
* Agriculture
* Mass media
* Health
* Government
* Energy

**Comment by argentlike** [comment]
Families first then. Then we build the branches ( ) and interdependencies <>
• Chemistry (chemicals, plastics)
• Mechanical engineering (mech components, cars, ships, trains)
• Electronics
• Extraction (mines, oil wells)
• Agriculture
• Mass media
• Health
• Government
• Energy

Becomes (DRAFT proposal)
• Chemistry (chemicals, plastics, nuclear)
• Mechanical engineering (mech components, cars, ships, trains, infrastructure)
• Electronics
• Primary Industries (mines, oil wells, agriculture, forestry)
• Health (medicine/medical/sports)
• Government (2x Secretariats: Socialist Secretariat (treasury/economics/finance) Party
Secretariat (control/propaganda/happiness/culture))
• Energy
• Mass media(do we need this separate?)

**Comment by argentlike** [comment]
Proposed we drop Mass Media
_do we need mass media separate or can we fit it into others with interdependencies?  We have mechanical, electronics, and govt.
It should all fit between these - or is there something arguably unique?
I would put it into the goverment family_

**Comment by argentlike** [comment]
New draft:
Becomes (DRAFT proposal)
• Chemistry (chemicals, plastics, nuclear)
• Mechanical engineering (mech components, cars, ships, trains, infrastructure)
• Electronics
• Primary Industries (mines, oil wells, agriculture, forestry)
• Health (medicine/medical/sports)
• Government (2x Secretariats: Socialist Secretariat (treasury/economics/finance) Party
Secretariat (control/propaganda/happiness/culture/mass media))
• Energy (fuel, )

**Comment by cit1zen** [comment]
I think that the families and streams make sense now. I am closing the discussion. The only thing left is to write a design document.
